survive-the-monsters.js

```diff
@@ -80,15 +86,81 @@ function normalizeInput() {
     input[0] /= m;
     input[1] /= m;
 }
-function applyMovement(dt) {
-    const speed = movementConfig.speed * dt;
+
+function applyPlayerMovement(dt) {
+    const speed = movementConfig.playerSpeed * dt;
     playerX += input[0] * speed;
     playerY += input[1] * speed;
 }
+function applyEnemiesMovement(dt) {
+    const speed = movementConfig.enemiesSpeed * dt;
+    
+    for (let enemyId = 0; enemyId <= enemiesLargestId; enemyId++) {
+        if (!enemiesState[enemyId]) {
+            continue;
+        }
+        const manDistX = playerX - enemiesX[enemyId];
+        const manDistY = playerY - enemiesY[enemyId];
+        const eucDistSq = manDistX**2 + manDistY**2;
+
+        let eucDist;
+        let eigX;
+        let eigY;
+        if (eucDistSq < 1) {
+            eucDist = 1;
+            eigX = manDistX;
+            eigY = manDistY;
+        } else {
+            eucDist = eucDistSq**0.5;
+            eigX = manDistX / eucDist;
+            eigY = manDistY / eucDist;
+        }
+
+        enemiesX[enemyId] += eigX * speed;
+        enemiesY[enemyId] += eigY * speed;
+    }
+}
+
 function updatePositions() {
```

Comment:
**issue**: Consider updating `enemiesLargestId` in the `despawnEnemy` function to ensure it reflects the current state of active enemies. This will help in managing the enemy array efficiently.

Reply:
done [here](https://github.com/grezisek1/survive-the-monsters/pull/5/commits/4e258a2457db035c5fe9c08c9b2119755842bd83#diff-10dfabc0f5ef0c2da531fd34c528ae230a2c5c30ec53097051c1d918873834be)